queue/driver_locker.go

```diff
@@ -108,42 +130,51 @@ func (l *lockManager) lockPinger(ctx context.Context) {
 				}
 
 				if op.ctx.Err() != nil {
+					grip.Debug(message.Fields{
+						"message":  "removing locally tracked lock",
+						"cause":    "context canceled",
+						"job_id":   name,
+						"stat":     stat,
+						"job_type": j.Type().Name,
+						"service":  "amboy.queue.locker",
+					})
 					delete(activeLocks, name)
 					continue
 				}
 
 				if err := l.d.SaveStatus(j, stat); err != nil {
+					grip.Debug(message.WrapError(err, message.Fields{
+						"message":  "problem updating lock",
+						"job_id":   name,
+						"stat":     stat,
+						"job_type": j.Type().Name,
+						"service":  "amboy.queue.locker",
+					}))
 					continue
 				}
 
 				activeLocks[name] = lockPingOp{
 					ts:  time.Now().Add(l.timeout / 2),
 					ctx: op.ctx,
 				}
-
-				timer.Reset(-time.Since(nextLoopAt))
 			}
+
+			timer.Reset(-time.Since(nextLoopAt))
 		}
 	}
 }
 
 func (l *lockManager) addPing(ctx context.Context, name string) {
 	wait := make(chan struct{})
-	payload := func(pings lockPings) {
+	l.ops <- func(pings lockPings) {
 		pings[name] = lockPingOp{
 			ts:  time.Now().Add(l.timeout),
 			ctx: ctx,
 		}
 		close(wait)
 	}
 
-	select {
-	case <-ctx.Done():
```

Comment:
Why no context check?

Reply:
the context is here to cancel the lock ping once it starts, not to cancel the act of adding a lock to the pinger. I think its less complicated to do this and was causing problems, I'd like to hold off on this change until later. just in terms of making this easier to update in evergreen.